File: 알고리즘/Find the Highest Altitude.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int pointer = 0;
        vector<int> a;
        a.push_back(0);
        for(auto ele : gain){
            pointer+=ele;
            a.push_back(pointer);
        }
        return *max_element(a.begin(), a.end());
    }   
};

File: 알고리즘/2/Find the Highest Altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        answer =[0];
        pointer = 0
        
        for ele in gain:
            pointer+=ele
            answer.append(pointer)
        return max(answer)
        
    def largestAltitude2(self, gain: List[int]) -> int:
        return max(accumulate(gain, initial=0))